lsp/java/src/main/java/com/itsaky/lsp/java/actions/common/ImportOrdererAction.java

```diff
@@ -0,0 +1,51 @@
+package com.itsaky.lsp.java.actions.common;
+
+import com.google.googlejavaformat.java.FormatterException;
+import com.google.googlejavaformat.java.ImportOrderer;
+import com.google.googlejavaformat.java.JavaFormatterOptions;
+import com.itsaky.androidide.actions.ActionData;
+import com.itsaky.androidide.app.BaseApplication;
+import com.itsaky.lsp.java.actions.BaseCodeAction;
+import com.itsaky.lsp.java.R;
+import io.github.rosemoe.sora.widget.CodeEditor;
+
+public class ImportOrdererAction extends BaseCodeAction {
+
+  @Override
+  public String getId() {
+    return "lsp_java_import_ordrerer_action";
+  }
+
+  @Override
+  public String getLabel() {
+    return BaseApplication.getBaseInstance().getApplicationContext().getString(getTitleTextRes());
+  }
+
+  @Override
+  protected int getTitleTextRes() {
+    return R.string.action_import_ordrerer;
+  }
+
+  @Override
+  public Boolean execAction(ActionData data) {
+    try {
+      CodeEditor editor = requireEditor(data);
+      String output = ImportOrderer.reorderImports(editor.getText().toString(), JavaFormatterOptions.Style.GOOGLE);
+      editor.setText(output);
+      return true;
+    } catch (FormatterException fe) {
+      fe.printStackTrace();
+      return false;
+    }
+  }
+
+  @Override
+  public boolean getRequiresUIThread() {
+    return true;
```

Comment:
This action must not be executed on UI thread. If there are a lot of import statements in the source code, it will block the UI thread.

Instead, order the imports in background thread and make the `executeAction` method return the "ordered" source code. The object returned from `execAction` is passed to `postExec` and `postExec` is always executed on UI thread.

So, in `postExec`, apply the returned source to editor.

See [`DefaultActionsRegistry.execInBackground`](https://github.com/itsaky/AndroidIDE/blob/70b94ec32c7c203c58cdf4e6399c683e25752a1e/actions/src/main/java/com/itsaky/androidide/actions/internal/DefaultActionsRegistry.kt#L149).